inc/dbmysql.class.php

```diff
@@ -791,43 +791,35 @@ function isSlave() {
     * @return boolean true if all query are successfull
     */
    function runFile($path) {
-      $DBf_handle = fopen($path, "rt");
-      if (!$DBf_handle) {
+      $script = fopen($path, 'r');
+      if (!$script) {
          return false;
       }
-
-      $formattedQuery = "";
-      $lastresult     = false;
-      while (!feof($DBf_handle)) {
-         // specify read length to be able to read long lines
-         $buffer = fgets($DBf_handle, 102400);
-
-         // do not strip comments due to problems when # in begin of a data line
-         $formattedQuery .= $buffer;
-         if ((substr(rtrim($formattedQuery), -1) == ";")
-             && (substr(rtrim($formattedQuery), -4) != "&gt;")
-             && (substr(rtrim($formattedQuery), -4) != "160;")) {
-
-            $formattedQuerytorun = $formattedQuery;
-
-            // Do not use the $DB->query
-            if ($this->query($formattedQuerytorun)) { //if no success continue to concatenate
-               $formattedQuery = "";
-               $lastresult     = true;
-               if (!isCommandLine()) {
-                  // Flush will prevent proxy to timeout as it will receive data.
-                  // Flush requires a content to be sent, so we sent sp&aces as multiple spaces
-                  // will be shown as a single one on browser.
-                  echo ' ';
-                  flush();
-               }
-            } else {
-               $lastresult = false;
+      $sql_query = @fread(
+         $script,
+         @filesize($path)
+      ) . "\n";
+      $sql_query = html_entity_decode($sql_query, ENT_COMPAT, 'UTF-8');
+
+      $sql_query = $this->removeSqlRemarks($sql_query);
+      $queries = preg_split('/;\s*$/m', $sql_query);
+
+      try {
+         $this->beginTransaction();
+         foreach ($queries as $query) {
+            $query = trim($query);
+            if ($query != '') {
+               $query = htmlentities($query);
+               $this->query($query);
             }
          }
+         $this->commit();
+      } catch (\Exception $e) {
+         $this->rollBack();
+         throw $e;
       }
```

Comment:
`DROP TABLE` and `CREATE TABLE` queries are implicitely commited and cannot be rolled back (see https://dev.mysql.com/doc/refman/8.0/en/implicit-commit.html).

I propose to quit at first error.
```suggestion
      foreach ($queries as $query) {
         $query = trim($query);
         if ($query != '') {
            $query = htmlentities($query);
            if (!$this->query($query)) {
               return false;
            }
         }
      }
```

Reply:
Okay, we have to check if the returned false is handled; script should really not continue (it was the point of throwing an exception)